### PR TITLE
Fix: Fix false negatives and reporting location in `require-meta-type` rule

### PR DESCRIPTION
**Pull request by bmish**
* Fixes false negatives
* Improves the reporting location

### CODE REVIEW

File: tests/lib/rules/require-meta-type.js [hunk 1]
```diff
@@ -99,7 +118,35 @@ ruleTester.run('require-meta-type', rule, {
...
+    },
+    {
+      code: `
+        const type = 'invalid-type';
+        module.exports = {
+          meta: { type },
+          create(context) {}
+        };
+      `,
+      errors: [{ messageId: 'unexpected', type: 'Identifier' }],
```
Comment (bmish) [inline tests/lib/rules/require-meta-type.js:131]: This test case was valid before (false negative).

File: tests/lib/rules/require-meta-type.js [hunk 2]
```diff
@@ -99,7 +118,35 @@ ruleTester.run('require-meta-type', rule, {
...
+      errors: [{ messageId: 'unexpected', type: 'Literal' }],
+    },
+    {
+      code: `
+        module.exports = {
+          meta: { type: undefined },
+          create(context) {}
+        };
+      `,
+      errors: [{ messageId: 'unexpected', type: 'Identifier' }],
```
Comment (bmish) [inline tests/lib/rules/require-meta-type.js:149]: This test case was valid before (false negative).

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks! 💯

**Comment by bmish** [comment]
@aladdin-add thanks! Several other rules have the same problems so I'll be submitting more PRs in the next few days.

**Pull request merged**